Sources/StreamChatUI/Components.swift

```diff
@@ -67,21 +67,88 @@ public struct _Components<ExtraData: ExtraDataTypes> {
 
     /// An object responsible for message layout options calculations in `ChatMessageListVC/ChatThreadVC`.
     public var messageLayoutOptionsResolver: _ChatMessageLayoutOptionsResolver<ExtraData> = .init()
+    
+    /// The view that shows a loading indicator.
+    public var loadingIndicator: ChatLoadingIndicator.Type = ChatLoadingIndicator.self
 
     // MARK: - Message list components
 
+    /// The collection view that shows the message list.
+    public var messageListCollectionView: ChatMessageListCollectionView<ExtraData>.Type = ChatMessageListCollectionView<ExtraData>
+        .self
+
+    /// The collection view layout used in `messageListCollectionView`.
+    public var messageListLayout: ChatMessageListCollectionViewLayout.Type = ChatMessageListCollectionViewLayout.self
+
+    /// The view that shows the date for currently visible messages on top of message list.
+    public var messageListScrollOverlayView: ChatMessageListScrollOverlayView.Type = ChatMessageListScrollOverlayView.self
+
     /// The view used to display content of the message, i.e. in the channel detail message list.
     public var messageContentView: _ChatMessageContentView<ExtraData>.Type = _ChatMessageContentView<ExtraData>.self
 
     /// The injector used to inject gallery attachment views.
     public var galleryAttachmentInjector: _AttachmentViewInjector<ExtraData>.Type = _GalleryAttachmentViewInjector<ExtraData>.self
+
     /// The injector used to inject link attachment views.
     public var linkAttachmentInjector: _AttachmentViewInjector<ExtraData>.Type = _LinkAttachmentViewInjector<ExtraData>.self
+
     /// The injector used for injecting giphy attachment views
     public var giphyAttachmentInjector: _AttachmentViewInjector<ExtraData>.Type = _GiphyAttachmentViewInjector<ExtraData>.self
+
     /// The injector used for injecting file attachment views
     public var filesAttachmentInjector: _FilesAttachmentViewInjector<ExtraData>.Type = _FilesAttachmentViewInjector<ExtraData>.self
 
+    /// The view that shows reactions bubble.
+    public var reactionsBubbleView: _ChatMessageReactionsBubbleView<ExtraData>.Type =
+        _ChatMessageDefaultReactionsBubbleView<ExtraData>.self
+
+    /// The view that shows reactions list in a bubble.
+    public var reactionsView: _ChatMessageReactionsView<ExtraData>.Type = _ChatMessageReactionsView<ExtraData>.self
+
+    /// The view that shows a single reaction.
+    public var reactionItemView: _ChatMessageReactionsView<ExtraData>.ItemView.Type =
+        _ChatMessageReactionsView<ExtraData>.ItemView.self
+
+    /// The view that shows author avatar in `messageContentView`.
+    public var messageAuthorAvatarView: ChatAvatarView.Type = ChatAvatarView.self
+
+    /// The view that shows thread participant avatar in `messageContentView`.
+    public var messageThreadParticipantAvatarView: ChatAvatarView.Type = ChatAvatarView.self
```

Comment:
This breaks the #1 rule of "UI Config Rules Re-Think":
> 1. The `UIConfig` inner structure should be flat and it should contain each view type just once.

Reply:
✅ Used `components.avatarView` instead